src/fides/api/ctl/sql_models.py

```diff
@@ -313,3 +315,65 @@ class SystemScans(Base):
     for _, sql_model in sql_model_map.items()
     if hasattr(sql_model, "is_default")
 ]
+
+
+class AllowedTypes(str, EnumType):
+    """Allowed types for custom field."""
+
+    string = "string"
+    string_list = "string[]"
+
+
+class ResourceTypes(str, EnumType):
+    """Resource types that can use custom fields."""
+
+    system = "system"
+    data_use = "data use"
+    data_category = "data category"
+    data_subject = "data subject"
+
+
+class CustomFieldDefinition(Base):
+    """Defines custom metadata for resources."""
+
+    __tablename__ = "plus_custom_field_definition"
+
+    fides_key = Column(String, index=True, unique=True)
+    name = Column(String, index=True, nullable=False)
+    description = Column(String)
+    field_type = Column(
+        EnumColumn(AllowedTypes),
+        nullable=False,
+    )
+    allow_list = Column(
+        ARRAY(String)
+    )  # Ulitmately this will come the the CustomList table
+    resource_type = Column(EnumColumn(ResourceTypes), nullable=False)
+    field_definition = Column(String, index=True)
+    custom_field = relationship(
+        "CustomField",
+        back_populates="plus_custom_field_definition",
```

Comment:
i think this should be the column/field name, rather than the table name, right?

Reply:
made a potential fix in https://github.com/ethyca/fides/pull/2212/files

Reply:
You are right, the change you did in #2212 looks correct so we can use the update there when we merge back here.